### ISSUE TITLE
🐛 Bug Report: GitLab Org ignores some groups

### ISSUE THREAD

**Issue by netvolart** [backstage/backstage#24914]
### 📜 Description

I've set up GitLab integration and discovery for a service catalog.
My gitlab groups have SAML integration with the Azure AD.

Works good at most of them. But It is ignoring some subgroups.

### 👍 Expected behavior

Sync all nested Gitlab groups and users with a mapping.

### 👎 Actual Behavior with Screenshots

This works perfectly:
I can see all groups, users with mappings, and relations between groups.
```
MainGroup/
└── FooGroup -> SAML links
    ├─ barGroup
    └─ bar2Group

```
And I can not sync this.
No CloudFoundation group, no PlatformTeam and SRETeam Groups.
And even I don't see any log messages related to them in Debug Mode. No errors. Just "processing" for previously existing groups.

I 've tried to drop catalog database and resync. Still ignoring this groups.

```
MainGroup/
└── CloudFoundation -> SAML links
    ├─ PlatformTeam -> SAML links
    └─ SRETeam -> SAML links
```

### 👟 Reproduction steps

```
  providers:
    gitlab:
      gitlab:
        host: gitlab.com
        orgEnabled: true
        group: maingroup 
        allowInherited: true 
        schedule: 
          frequency: { minutes: 15 }
          timeout: { minutes: 15 }
        rules:
          - allow: [Group, User]
```

### 📃 Provide the context for the Bug.

No related logs. Total ignorance for this group

### 🖥️ Your Environment

```
OS:   Darwin 23.0.0 - darwin/arm64
node: v20.9.0
yarn: 1.22.19
cli:  0.26.5 (installed)
backstage:  1.27.2

Dependencies:
  @backstage/app-defaults                                          1.5.5
  @backstage/backend-app-api                                       0.7.5
  @backstage/backend-common                                        0.21.7, 0.22.0
  @backstage/backend-defaults                                      0.2.18
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.11
  @backstage/backend-plugin-api                                    0.6.18
  @backstage/backend-tasks                                         0.5.23
  @backstage/catalog-client                                        1.6.5
  @backstage/catalog-model                                         1.5.0
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.5
  @backstage/cli                                                   0.26.5
  @backstage/config-loader                                         1.8.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.5
  @backstage/core-compat-api                                       0.2.5
  @backstage/core-components                                       0.14.7
  @backstage/core-plugin-api                                       1.9.2
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.8
  @backstage/frontend-plugin-api                                   0.6.5
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.27
  @backstage/integration                                           1.11.0
  @backstage/plugin-api-docs                                       0.11.5
  @backstage/plugin-app-backend                                    0.3.67
  @backstage/plugin-app-node                                       0.1.18
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.10
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.10
  @backstage/plugin-auth-backend-module-azure-easyauth-provider    0.1.1
  @backstage/plugin-auth-backend-module-bitbucket-provider         0.1.1
  @backstage/plugin-auth-backend-module-cloudflare-access-provider 0.1.1
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.13
  @backstage/plugin-auth-backend-module-github-provider            0.1.15
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.15
  @backstage/plugin-auth-backend-module-google-provider            0.1.15
  @backstage/plugin-auth-backend-module-guest-provider             0.1.4
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.13
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.15
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.11
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.9
  @backstage/plugin-auth-backend-module-okta-provider              0.0.11
  @backstage/plugin-auth-backend                                   0.22.5
  @backstage/plugin-auth-node                                      0.4.13
  @backstage/plugin-auth-react                                     0.1.2
  @backstage/plugin-catalog-backend-module-gitlab-org              0.0.1
  @backstage/plugin-catalog-backend-module-gitlab                  0.3.15
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.16
  @backstage/plugin-catalog-backend                                1.22.0
  @backstage/plugin-catalog-common                                 1.0.23
  @backstage/plugin-catalog-graph                                  0.4.5
  @backstage/plugin-catalog-import                                 0.11.0
  @backstage/plugin-catalog-node                                   1.12.0
  @backstage/plugin-catalog-react                                  1.12.0
  @backstage/plugin-catalog                                        1.20.0
  @backstage/plugin-events-backend-module-gitlab                   0.2.4
  @backstage/plugin-events-backend                                 0.3.5
  @backstage/plugin-events-node                                    0.3.4
  @backstage/plugin-home-react                                     0.1.13
  @backstage/plugin-home                                           0.7.4
  @backstage/plugin-org                                            0.6.25
  @backstage/plugin-permission-backend-module-allow-all-policy     0.1.15
  @backstage/plugin-permission-backend                             0.5.42
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.29
  @backstage/plugin-permission-react                               0.4.22
  @backstage/plugin-proxy-backend                                  0.4.16
  @backstage/plugin-scaffolder-backend-module-azure                0.1.10
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.8
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.10
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.8
  @backstage/plugin-scaffolder-backend-module-github               0.2.8
  @backstage/plugin-scaffolder-backend-module-gitlab               0.4.0
  @backstage/plugin-scaffolder-backend                             1.22.7
  @backstage/plugin-scaffolder-common                              1.5.2
  @backstage/plugin-scaffolder-node                                0.4.4
  @backstage/plugin-scaffolder-react                               1.8.5
  @backstage/plugin-scaffolder                                     1.20.0
  @backstage/plugin-search-backend-module-catalog                  0.1.24
  @backstage/plugin-search-backend-module-techdocs                 0.1.23
  @backstage/plugin-search-backend-node                            1.2.23
  @backstage/plugin-search-backend                                 1.5.9
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.11
  @backstage/plugin-search                                         1.4.11
  @backstage/plugin-techdocs-backend                               1.10.5
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.10
  @backstage/plugin-techdocs-node                                  1.12.4
  @backstage/plugin-techdocs-react                                 1.2.4
  @backstage/plugin-techdocs                                       1.10.5
  @backstage/plugin-user-settings                                  0.8.6
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.9.0
  @backstage/test-utils                                            1.5.5
  @backstage/theme                                                 0.5.5
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.8
✨  Done in 1.10s.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by elaine-mattos** [comment]
Hi @netvolart ,

do the GitLab groups that are not showing have direct members? If not and you still want to show these groups, you can use the option [`allowInherited: true`](https://backstage.io/docs/integrations/gitlab/org#configuration) to force the ingestion process to also add these groups.

**Comment by netvolart** [comment]
> Hi @netvolart ,
> 
> do the GitLab groups that are not showing have direct members? If not and you still want to show these groups, you can use the option [`allowInherited: true`](https://backstage.io/docs/integrations/gitlab/org#configuration) to force the ingestion process to also add these groups.

Hi, @elaine-mattos 
The group doesn't have direct members. But I want to sync it. SAML users are not in sync if I am not adding them directly to the group using GitLab console?

allowInherited: true option configured. Not syncing, unfortunately

**Comment by netvolart** [comment]
I've found a problem. @elaine-mattos 

- Backstage is syncing only Direct Members (It's correct because otherwise group will have all inherited users)
- SAML users become a "Direct Member" if:
 1) User SAML group has permissions level NOT less than upper level group.

For example: If the user is a `Maintainer` on a "MainGroup" level, it should be in an AD group mapped to the `Maintainer` role on a PlatformTeam level.
My confusion was because SRETeam and PlatformTeam users are maintainers, but I've added them as `Developers` in nested groups.
 
2) SAML user should logout and login. After that GitLab will show him as a SAML:    Direct Member. 

The only one,not critical, thing - Backstage is not syncing empty Groups (without direct members) even with `allowInherited: true`
